clickhouse/tests/test_clickhouse.py

```diff
@@ -0,0 +1,63 @@
+# (C) Datadog, Inc. 2019
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import pytest
+
+from datadog_checks.clickhouse import ClickhouseCheck
+
+from .common import CLICKHOUSE_VERSION
+from .metrics import ALL_METRICS
+
+pytestmark = [pytest.mark.integration, pytest.mark.usefixtures('dd_environment')]
+
+
+def test_check(aggregator, instance):
+    check = ClickhouseCheck('clickhouse', {}, [instance])
+    check.run()
+
+    server_tag = 'server:{}'.format(instance['server'])
+    port_tag = 'port:{}'.format(instance['port'])
+    for metric in ALL_METRICS:
+        aggregator.assert_metric_has_tag(metric, server_tag)
+        aggregator.assert_metric_has_tag(metric, port_tag)
+        aggregator.assert_metric_has_tag(metric, 'db:default')
+        aggregator.assert_metric_has_tag(metric, 'foo:bar')
+
+    aggregator.assert_metric('clickhouse.table.replicated.total', 2)
+    aggregator.assert_metric(
+        'clickhouse.dictionary.item.current', tags=[server_tag, port_tag, 'db:default', 'foo:bar', 'dictionary:test']
+    )
+
+
+def test_custom_queries(aggregator, instance):
+    instance['custom_queries'] = [
+        {
+            'tags': ['test:clickhouse'],
+            'query': 'SELECT COUNT(*) FROM system.settings WHERE changed',
+            'columns': [{'name': 'settings.changed', 'type': 'gauge'}],
+        }
+    ]
+
+    check = ClickhouseCheck('clickhouse', {}, [instance])
+    check.run()
+
+    aggregator.assert_metric(
+        'clickhouse.settings.changed',
+        metric_type=0,
+        tags=[
+            'server:{}'.format(instance['server']),
+            'port:{}'.format(instance['port']),
+            'db:default',
+            'foo:bar',
+            'test:clickhouse',
+        ],
+    )
+
+
+@pytest.mark.skipif(CLICKHOUSE_VERSION == 'latest', reason='Version `latest` is ever-changing, skipping')
+def test_version_metadata(aggregator, instance, datadog_agent):
+    check = ClickhouseCheck('clickhouse', {}, [instance])
+    check.check_id = 'test:123'
+    check.run()
+
+    datadog_agent.assert_metadata('test:123', {'version.scheme': 'calver', 'version.year': CLICKHOUSE_VERSION})
```

Comment:
Could you add `datadog_agent.assert_metadata_count` to check parts other than `year` are present ?

Reply:
It needs an `at_least` option or else it'll be flaky

https://github.com/DataDog/integrations-core/blob/30ea596677c58934b7fcca940cd3338fa69490ab/clickhouse/datadog_checks/clickhouse/clickhouse.py#L66-L67